### PR TITLE
Issue-10-Make-readable-file-name

### PR DESCRIPTION
**Pull request by bharattech**
Used the title of the page when available to make the file names more easily readable.

### CODE REVIEW

**Comment by bharattech** [comment]
Attached is a screenshot capturing the results of the performed test.
Screenshot from 2023-12-21 20-38-03

**Comment by bharattech** [comment]
@misslivirose , could you please review this pull request? Your feedback is valuable.

File: extension/popup/memory_cache.js [hunk 1]
```diff
@@ -3,23 +3,40 @@ const DOWNLOAD_SUBDIRECTORY = "MemoryCache";
 /*
 Generate a file name based on date and time
 */
-function generateFileName(ext) {
-  return (
-    new Date().toISOString().concat(0, 19).replaceAll(":", ".") + "." + ext
-  );
+
+function getTabTitle() {}
```
Comment (zfox23) [inline extension/popup/memory_cache.js:7]: it appears as though this function is unused and extraneous. if that's the case, could you please remove it?

File: extension/popup/memory_cache.js [hunk 2]
```diff
@@ -3,23 +3,40 @@ const DOWNLOAD_SUBDIRECTORY = "MemoryCache";
...
+      }
+    })
+    .catch((error) => {
+      reject(`Error querying tabs: ${error}`);
+    });
 }
 
 async function savePDF() {
   try {
+    var fileName = await generateFileName("pdf");
```
Comment (zfox23) [inline extension/popup/memory_cache.js:30]:
```suggestion
    let fileName = await generateFileName("pdf");
```

**Review by misslivirose — APPROVED** [review]

**Pull request merged**